helpers/arrays.tsx

```diff
@@ -18,3 +18,17 @@ export function sortObjectsByParam(
     return a[param] < b[param] ? -1 : 1;
   });
 }
+
+export function filterObjectsByParams(
+  ary: Array<any>,
+  params: { [key: string]: string }
+): Array<any> {
+  return ary.filter(function (obj: any) {
+    // find any params that match the search terms
+    return (
+      Object.keys(params).filter((key) =>
```

Comment:
[concern] I mentioned this before, but I suspect we will need something a little more robust as we go along, to accommodate uppercase vs lowercase, similar characters like é and e, etc, although for our immediate purposes I think this will be fine!  

One kinda tricky edge case, which I introduced by throwing a label in there, is that for users with no name (I would have to look to remember if they have an undefined username or just an empty string), they're currently being labeled "Unnamed" in the interface, but the search here doesn't turn that up since the username / presentation names don't reflect the papered over label.  Might be worth a future issue to think about how we want to actually display or flag unnamed users so that they make sense in combination with the search feature?

Reply:
It occurs to me that we'll have a similar problem with the search if we're swapping out guid names for service account key names and things of that nature, as well.  I wonder if ultimately it makes sense to alter the user object so that there's a search field and we can combine things like username / presentation name / other display names and whatever fields we think would be good to search, then use that for the search?  Basically like a lucene "text" dump field.  I'm solutionizing more than we need for this PR, though.

Reply:
That's a good question, we may need to ask some cloud folks what we can expect in that environment.  Let's not worry about it for now!